File: 2020 listopad/c++/2.cpp
#include <iostream>
#include <fstream>
#include <list>
#include <set>
using namespace std;

list <int> lucky;
set <int> s;

int main()
{
    ifstream in("dane.txt");
    
    for (int i = 1; i <= 10000; i += 2)
        lucky.push_back(i);
    
    list <int>::iterator cur = lucky.begin();
    cur++;
    while (cur != lucky.end())
    {
        int l = *cur, i = 1;
        for (list <int>::iterator it = lucky.begin(); it != lucky.end(); it++)
        {
            if (i++ % l == 0)
            {
                list <int>::iterator tmp = --it;
                it++;
                lucky.erase(it);
                it = tmp;
            }
        }
        cur++;
    }

    for (int v: lucky)
        s.insert(v);
    
    int n;
    pair <int, int> ans = {0, 0}, akt = {0, 0};
    bool was = false;
    while (in >> n)
    {
        if (s.count(n))
        {
            if (was)
                akt.first++;
            else
                akt = {1, n};

            if (akt.first > ans.first)
                ans = akt;

            was = true;
        }
        else
        {
            akt = {0, 0};
            was = false;
        }
    }   
    cout << ans.first << ' ' << ans.second << '\n';
}

File: 2020 listopad/python/2.py
with open("dane.txt", "r") as f:
    dane = list(map(int, f.read().strip().split('\n')))

lucky = [i for i in range(1, 10001, 2)]
cur = 1
while cur < len(lucky):
    crossed = [lucky[i] for i in range(lucky[cur] - 1, len(lucky), lucky[cur])]
    for i in crossed:
        lucky.remove(i)
    cur += 1
lucky = set(lucky)

ans, cur, was = [0, 0], [0, 0], False
for i in dane:
    if i in lucky:
        if was:
            cur[0] += 1
        else:
            cur = [1, i]
        
        if cur[0] > ans[0]:
            ans = cur

        was = True
    else:
        cur = (0, 0)
        was = False
        
print(*ans)